pkg/schemaconv/testdata/swagger.json

```diff
@@ -78578,6 +78578,12 @@
    },
    "io.k8s.api.apps.v1beta1.DeploymentStrategy": {
     "description": "DeploymentStrategy describes how to replace existing pods with new ones.",
+    "x-kubernetes-unions": [{
+     "discriminator": "type",
+     "fields-discriminated": {
```

Comment:
does this mean we don't support non-discriminator unions? Would rather call it `fields` instead of `fields-discriminated`.

Reply:
as the type is different, we would in fact need `fields []string` and `discriminatedFields map[string]string`.

Reply:
Yeah I'll probably do that ... if `discriminator` is set, then `fields` must be unset and `fields-discriminated` must be set. and the other way around.